.drone.yml

```diff
@@ -38,38 +38,38 @@ steps:
       - tox -e missing-migrations
 
   - name: tests
-    image: abakus/lego-testbase:python3.9
+    image: abakus/lego-testbase:poetry
     when:
-      event: push
+      event: [push]
     environment:
       LANG: C.UTF-8
       DATABASE: database
     commands:
       - tox -e tests
 
   - name: isort
-    image: abakus/lego-testbase:python3.9
+    image: abakus/lego-testbase:poetry
     when:
-      event: push
+      event: [push]
     environment:
       LANG: C.UTF-8
     commands:
       - tox -e isort
 
   - name: black
-    image: abakus/lego-testbase:python3.9
-    pull: true
+    image: abakus/lego-testbase:poetry
+    pull: if-not-exists
```

Comment:
Any reason this is only added to this step? Just genuinely wondering

Reply:
Hmm good question!

The drone docs aren't very clear on what the default value of pull is imo, but the way I understand the docs, we wouldn't really need to use the `pull` attribute as long as we're not tagging out images with `:latest`, unless we do not want it to cache.  
So the default behavior for non-latest tags appears to be if-not-exists.

But regarding why it's just on this step, I suppose it is because this file was originally copied from `lego`, and in https://github.com/webkom/lego/blob/master/.drone.yml#L69-L92 the only two steps that have the `pull` attribute set are `black` and `mypy`, and mypy is not implemented in admissions.

`pull: true` does not appear to be a valid tag however, but if it is processed at all I would guess it reads as `pull: always`, which would mean unnecessary fetching of docker images that are cached

Reply:
Docs for reference: https://docs.drone.io/pipeline/docker/syntax/images/#pulling-images

Reply:
Yes I see. I've had mixed experiences with Drone's "cache", so imo. it's fine to leave it.

Reply:
Fair enough. Although I guess we might eventually decide on whether or not we type out the configuration or not - as all the steps *should* now be configured the same way, but `black` is the only one where it's written out